include/seastar/core/semaphore.hh

```diff
@@ -308,6 +332,16 @@ public:
     void ensure_space_for_waiters(size_t n) {
         _wait_list.reserve(n);
     }
+
+#ifdef SEASTAR_DEADLOCK_DETECTION
```

Comment:
Method names should be more verbose about the purpose they serve. Also, imo if they serve as public methods, they should have documentation provided. In that case, maybe it's better to simply befriend `semaphore_units` and make these two methods private.